### DISCUSSION TITLE
Menu opened by button without 'on_click' handler

### DISCUSSION THREAD

**Discussion by thetableman** [zauberzeug/nicegui#2000] [Q&A] [answered]
### Question

Not sure if this is a bug or an intended change I've missed in the documentation (I've been away a few weeks and am still catching up).

The below allows opening and closing of the menu without an `on_click` handler provided:
```py
from nicegui import ui

with ui.button('Click Me') as button: 
    with ui.menu() as menu:
        ui.menu_item('Item 1')

ui.run()
```

The nested `with` reference to `ui.menu()` seems to automatically imply the button handler is to open/close the menu. Whilst this is more than likely in most scenarios I'm not sure this is the preferred outcome.

**Comment by falkoschindler** [comment, marked as answer]
Hi @thetableman, this has been changed in version 1.3.0 to match the default behavior of Quasar's menu component.
See https://github.com/zauberzeug/nicegui/issues/713 for a more detailed explanation.
